programs/stake/src/stake_state.rs

```diff
@@ -263,54 +263,81 @@ impl Delegation {
         }
     }
 
-    fn stake_and_activating(&self, epoch: Epoch, history: Option<&StakeHistory>) -> (u64, u64) {
+    fn stake_and_activating(
+        &self,
+        target_epoch: Epoch,
+        history: Option<&StakeHistory>,
+    ) -> (u64, u64) {
         if self.is_bootstrap() {
+            // fully effective immediately
             (self.stake, 0)
-        } else if epoch == self.activation_epoch {
+        } else if target_epoch == self.activation_epoch {
+            // all is activating
             (0, self.stake)
-        } else if epoch < self.activation_epoch {
+        } else if target_epoch < self.activation_epoch {
+            // not yet enabled
             (0, 0)
-        } else if let Some((history, mut entry)) = history.and_then(|history| {
-            history
-                .get(&self.activation_epoch)
-                .map(|entry| (history, entry))
-        }) {
-            // && !is_bootstrap() && epoch > self.activation_epoch
-            let mut effective_stake = 0;
-            let mut next_epoch = self.activation_epoch;
+        } else if let Some((history, mut prev_epoch, mut prev_cluster_stake)) =
+            history.and_then(|history| {
+                history
+                    .get(&self.activation_epoch)
+                    .map(|cluster_stake_at_activation_epoch| {
+                        (
+                            history,
+                            self.activation_epoch,
+                            cluster_stake_at_activation_epoch,
+                        )
+                    })
+            })
+        {
+            // target_epoch > self.activation_epoch
 
-            // loop from my activation epoch until the current epoch
-            //   summing up my entitlement
+            // loop from my activation epoch until the target epoch summing up my entitlement
+            // current stake is updated using its previous epoch's cluster stake
+            let mut current_epoch;
+            let mut current_effective_stake = 0;
```

Comment:
a bit of delicacy, but these `current_` grouped variables are rhymed with these `prev_`s: https://github.com/solana-labs/solana/pull/13460/files#diff-ab7997cf6770a6fae7b258e433b5dd8c33afa30431eab212552bc7f2b7b38db1R280